### PR TITLE
Adding dynamic templates for device types listing

### PR DESCRIPTION
**Pull request by rasika90**

### CODE REVIEW

File: features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
-    <td class="fade-edge remove-padding-top" data-search="{{deviceType.name}}" data-display="{{deviceType.name}}"
-        data-grid-label="Type">{{deviceType.name}}</td>
+    <td class="fade-edge remove-padding-top" data-search="{{deviceTypeLabel}}" data-display="{{deviceTypeLabel}}"
+        data-grid-label="Type">{{deviceTypeLabel}}</td>
     <td class="text-right content-fill text-left-on-grid-view no-wrap">
-        <a href="{{../appContext}}/device/{{deviceType.name}}/enroll"
-           data-click-event="remove-form" class="btn padding-reduce-on-grid-view">
+        <a href="{{appContext}}/device/{{deviceTypeName}}/enroll"
+           data-click-event="remove-form" class="btn col-md-12">
                             <span class="fw-stack">
```
Comment (milanperera) [inline features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs]: Fix indentation issue

File: features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        <div class="thumbnail icon">
+            <i class="square-element text fw fw-android"></i>
+        </div>
+    </td>
+    <td class="fade-edge remove-padding-top" data-search="{{deviceTypeLabel}}" data-display="{{deviceTypeLabel}}"
+        data-grid-label="Type">{{deviceTypeLabel}}</td>
+    <td class="text-right content-fill text-left-on-grid-view no-wrap">
+        <a href="{{appContext}}/device/{{deviceTypeName}}/enroll"
+           data-click-event="remove-form" class="btn col-md-12">
+                            <span class="fw-stack">
```
Comment (milanperera) [inline features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs]: Fix indentation issue.

File: features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <div class="thumbnail icon">
+            <i class="square-element text fw fw-windows"></i>
+        </div>
+    </td>
+    <td class="fade-edge remove-padding-top" data-search="{{deviceTypeLabel}}" data-display="{{deviceTypeLabel}}"
+        data-grid-label="Type">{{deviceTypeLabel}}</td>
+    <td class="text-right content-fill text-left-on-grid-view no-wrap">
+        <a href="{{appContext}}/device/{{deviceTypeName}}/enroll"
+           data-click-event="remove-form" class="btn col-md-12">
+                            <span class="fw-stack">
```
Comment (milanperera) [inline features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs]: Fix indentation issue.

File: features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs [hunk 4]
```diff
@@ -1,30 +1,19 @@
-<tr data-type="selectable" data-deviceid="{{deviceType.id}}" data-devicetype="{{deviceType.name}}">
```
Comment (DilanUA) [inline features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs]: no license header seen

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs [hunk 5]
```diff
@@ -39,7 +41,7 @@
                                        placeholder="Ex. Lobby_Firealarm"
                                        name="deviceName" size="60" required>
```
Comment (DilanUA) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: fix indentation formatting issues of this file

File: features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<tr data-type="selectable" data-deviceid="{{deviceTypeId}}" data-devicetype="{{deviceTypeName}}">
```
Comment (DilanUA) [inline features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs]: no license header seen

File: features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<tr data-type="selectable" data-deviceid="{{deviceTypeId}}" data-devicetype="{{deviceTypeName}}">
```
Comment (DilanUA) [inline features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs]: no license header seen

File: features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs (hunk 1 above)
Comment (rasika) [inline features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

File: features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs (hunk 2 above)
Comment (rasika) [inline features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

File: features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs (hunk 3 above)
Comment (rasika) [inline features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

File: features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs (hunk 4 above)
Comment (rasika) [inline features/device-mgt-iot-androidsense-feature/org.wso2.carbon.device.mgt.iot.androidsense.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android_sense.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs (hunk 5 above)
Comment (rasika) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: fixed in 39d2ce12dc1eba6893ad42b3dd307435bb73144c

File: features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs (hunk 6 above)
Comment (rasika) [inline features/device-mgt-mdm-android-feature/org.wso2.carbon.device.mgt.mobile.android.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.android.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

File: features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs (hunk 7 above)
Comment (rasika) [inline features/device-mgt-mdm-windows-feature/org.wso2.carbon.device.mgt.mobile.windows.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.windows.type-view/public/templates/listing.hbs]: this file does not exist with 39d2ce12dc1eba6893ad42b3dd307435bb73144c. Hence cannot fix.

**Pull request merged**